langchain4j-core/src/main/java/dev/langchain4j/model/discovery/ModelDescription.java

```diff
@@ -0,0 +1,319 @@
+package dev.langchain4j.model.discovery;
+
+import dev.langchain4j.model.ModelProvider;
+import dev.langchain4j.model.chat.Capability;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Represents metadata about an available model from a provider.
+ * This class provides a unified view of model information across different providers.
+ *
+ * <p>Only {@code id}, {@code name}, and {@code provider} are required fields.
+ * All other fields are optional and may be null depending on what information
+ * the provider makes available.
+ */
+public class ModelDescription {
+
+    private final String id;
+    private final String name;
+    private final String description;
+    private final ModelProvider provider;
+    private final ModelType type;
+    private final Set<Capability> capabilities;
+    private final ModelPricing pricing;
+    private final Integer contextWindow;
+    private final Integer maxOutputTokens;
+    private final Instant createdAt;
+    private final String owner;
+    private final Boolean deprecated;
+    private final Set<String> supportedLanguages;
+    private final Map<String, Object> additionalMetadata;
+
+    private ModelDescription(Builder builder) {
+        this.id = Objects.requireNonNull(builder.id, "id must not be null");
+        this.name = Objects.requireNonNull(builder.name, "name must not be null");
+        this.provider = Objects.requireNonNull(builder.provider, "provider must not be null");
+        this.description = builder.description;
+        this.type = builder.type;
+        this.capabilities = builder.capabilities != null
+                ? Collections.unmodifiableSet(new HashSet<>(builder.capabilities))
+                : Set.of();
+        this.pricing = builder.pricing;
+        this.contextWindow = builder.contextWindow;
+        this.maxOutputTokens = builder.maxOutputTokens;
+        this.createdAt = builder.createdAt;
+        this.owner = builder.owner;
+        this.deprecated = builder.deprecated;
+        this.supportedLanguages = builder.supportedLanguages != null
```

Comment:
```suggestion
        this.supportedLanguages = copy(builder.supportedLanguages);
```
pleause use internal utils everywhere